src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/MidEventCallback.java

```diff
@@ -69,13 +70,12 @@ public JsonNode invoke(String caseTypeId,
                 if (StringUtils.isNotEmpty(content.getCaseReference())) {
                     CaseDetails caseDetails = caseService.getCaseDetails(caseType.getJurisdictionId(), content.getCaseReference());
                     caseDetailsBefore = caseService.clone(caseDetails);
-                    currentOrNewCaseDetails = caseService.populateCurrentCaseDetailsWithEventFields(content,
-                        caseDetails);
-
+                    currentOrNewCaseDetails = caseService.populateCurrentCaseDetailsWithEventFields(content, caseDetails);
                 } else {
                     currentOrNewCaseDetails = caseService.createNewCaseDetails(caseTypeId, caseType.getJurisdictionId(),
                         content.getEventData() == null ? content.getData() : content.getEventData());
                 }
+                removeNextPageFieldData(currentOrNewCaseDetails, wizardPageOptional.get().getOrder(), caseTypeId, event.getEventId());
```

Comment:
If I don't recall wrong, page display order is optional. Even though the ordering is guaranteed only if a display order is defined, if it's missing I think pages are ordered by page id or in the order they are defined on the spreadsheet (can't recall exactly). In this case is it possible that this code will remove fields from a previous page?
Worth doing some testing.

Reply:
Will check this one...